hooks/cascading-scans/hook.ts

```diff
@@ -39,18 +42,26 @@ async function getCascadingRules(scan: Scan): Promise<Array<CascadingRule>> {
   return <Array<CascadingRule>>await getCascadingRulesForScan(scan);
 }
 
+async function getCascadedRule(scan: Scan): Promise<CascadingRule> {
+  // Explicit Cast to the proper Type
+  return <CascadingRule> await getCascadedRuleForScan(scan);
+}
+
 /**
  * Goes thought the Findings and the CascadingRules
  * and returns a List of Scans which should be started based on both.
  */
 export function getCascadingScans(
   parentScan: Scan,
   findings: Array<Finding>,
-  cascadingRules: Array<CascadingRule>
+  cascadingRules: Array<CascadingRule>,
+  cascadedRule: CascadingRule
 ): Array<ExtendedScanSpec> {
   let cascadingScans: Array<ExtendedScanSpec> = [];
   const cascadingRuleChain = getScanChain(parentScan);
 
+  parentScan = purgeCascadedRuleFromScan(parentScan, cascadedRule);
```

Comment:
This took me a minute to understand why this is needed.
I think this should at least have a comment explaining why this is required or be moved closer to the point where the env vars & volumes are merged to make it clear that this is done to avoid duplicates there.

Reply:
I've added some comments. Feel free to add more or move it where you think it fits better.